lib/std/math/isnormal.zig

```diff
@@ -1,32 +1,20 @@
 const std = @import("../std.zig");
 const math = std.math;
 const expect = std.testing.expect;
-const maxInt = std.math.maxInt;
 
 // Returns whether x has a normalized representation (i.e. integer part of mantissa is 1).
 pub fn isNormal(x: anytype) bool {
     const T = @TypeOf(x);
-    switch (T) {
-        f16 => {
-            const bits = @bitCast(u16, x);
-            return (bits + (1 << 10)) & (maxInt(u16) >> 1) >= (1 << 11);
-        },
-        f32 => {
-            const bits = @bitCast(u32, x);
-            return (bits + (1 << 23)) & (maxInt(u32) >> 1) >= (1 << 24);
-        },
-        f64 => {
-            const bits = @bitCast(u64, x);
-            return (bits + (1 << 52)) & (maxInt(u64) >> 1) >= (1 << 53);
-        },
-        f128 => {
-            const bits = @bitCast(u128, x);
-            return (bits + (1 << 112)) & (maxInt(u128) >> 1) >= (1 << 113);
-        },
-        else => {
-            @compileError("isNormal not implemented for " ++ @typeName(T));
-        },
+
+    if (@typeInfo(T) != .Float) {
+        @compileError("isNormal not implemented for " ++ @typeName(T));
     }
+
+    const IntT = std.meta.Int(.unsigned, @bitSizeOf(T));
+    const overflow = 1 << math.floatMantissaBits(T);
+    const remove_sign = math.maxInt(IntT) >> 1;
+
+    return (@bitCast(IntT, x) +% overflow) & remove_sign >= (overflow << 1);
```

Comment:
Declutter this abit. Also add a brief and comprehensive comment why the lhs works.
You mentioned `hack`, so readers can not validate its correctness/follow easily.
```zig
// your cool explanation
var lhs = (@bitCast(IntT, x) +% overflow) & remove_sign;
return lhs >= (overflow << 1);
```